src/main/kotlin/com/charleskorn/kaml/YamlOutput.kt

```diff
@@ -41,10 +44,19 @@ import java.util.Optional
 
 internal class YamlOutput(
     writer: StreamDataWriter,
-    private val configuration: YamlConfiguration
+    private val yaml: Yaml
 ) : ElementValueEncoder() {
     private val settings = DumpSettingsBuilder().build()
     private val emitter = Emitter(settings, writer)
+    private var currentTag: String? = null
+    private val configuration: YamlConfiguration get() = yaml.configuration
+    override val context: SerialModule get() = yaml.context
+    private val yamlTag: Optional<String>
```

Comment:
Is there a reason why this property mutates the state? Maybe instead of this, in `beginStructure`, we can construct the `Optional.ofNullable(...)` value and then set `currentTag` to `null`? Or maybe turn this property into a method that does what this getter is doing?